samza-core/src/main/scala/org/apache/samza/container/TaskInstance.scala

```diff
@@ -453,6 +472,11 @@ class TaskInstance(
                 "during async stage of commit for taskName: %s checkpointId: %s. New exception logged above. " +
                 "Saved exception under Caused By.", commitException.get())
             }
+          } else {
+            metrics.commitNs.update(System.nanoTime() - commitStartNs)
+            // reset the numbers skipped commits for the current commit
+            numSkippedCommits = 0
```

Comment:
It may not be correct to reset this gauge since the default reporting interval is only 1 minute. E.g., if you set task.commit.ms = 10 secs and skip 5 commits, the reported gauge value may still incorrectly read 0 at the end. Better to leave this as an incrementing only value.